lib/primitives/covenant.js

```diff
@@ -529,12 +529,69 @@ class Covenant extends bio.Struct {
     return bw.render();
   }
 
+  /**
+   * Render the Covenant as human
+   * readable assembly.
+   * @returns {String}
+   */
+
+  toASM() {
+    const schema = [
```

Comment:
This is not extensible to future and unknown covenant types. The schema needs to be explicit for each type, and render something like `UNKNOWN` when an unknown covenant version is found.

Reply:
I also think that such a schema should live in a `common` file